### PR TITLE
Improve `data` and `fraction` args across export formats

### PR DESCRIPTION
**Pull request by lakshanthad**
## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
📘 This PR updates YOLO26 export support and documentation to better reflect which formats support INT8 calibration options like `data` and `fraction`, improving consistency between code, docs, and tutorial tables.

### 📊 Key Changes
- Added `data` and `fraction` export arguments to the docs for:
  - TensorFlow SavedModel
  - TensorFlow.js
  - TensorFlow Edge TPU
- Updated the shared export format tables in:
  - `docs/macros/export-table.md`
  - `examples/tutorial.ipynb`
  so the listed arguments now match actual exporter capabilities more closely.
- Updated `ultralytics/engine/exporter.py` export format definitions:
  - **SavedModel** now includes `fraction`
  - **TF.js** now includes `fraction`
  - **Edge TPU** now supports `int8`, `data`, and `fraction`
- Cleaned up a few incorrect or overly broad argument listings:
  - **CoreML** no longer lists `data`
  - **MNN** no longer lists `data`

### 🎯 Purpose & Impact
- ✅ Makes export documentation more accurate, so users know which options are actually available for each deployment target.
- 🚀 Improves support for INT8 quantization workflows by exposing calibration-related settings like dataset selection (`data`) and subset control (`fraction`) in more TensorFlow-based exports.
- 🧩 Reduces confusion caused by mismatches between docs, tutorials, and implementation.
- 📱 Helps users targeting edge and browser deployments choose the right export settings with more confidence.
- 🔧 For developers, this means fewer surprises when exporting models and better alignment between the API and the published docs.

### CODE REVIEW

**Comment by lakshanthad** [comment]
@Laughing-q This is an extension of this PR:
https://github.com/ultralytics/ultralytics/pull/24362

**Comment by UltralyticsAssistant** [comment]
👋 Hello @lakshanthad, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This is an automated message to help with review readiness, and an engineer will assist you shortly. Please review the checklist below to help ensure a smooth integration. 😊

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

File: ultralytics/engine/exporter.py [hunk 1]
```diff
@@ -139,14 +139,14 @@ def export_formats():
...
         ],
-        ["CoreML", "coreml", ".mlpackage", True, False, ["batch", "data", "dynamic", "half", "int8", "nms"]],
-        ["TensorFlow SavedModel", "saved_model", "_saved_model", True, True, ["batch", "data", "int8", "keras", "nms"]],
+        ["CoreML", "coreml", ".mlpackage", True, False, ["batch", "dynamic", "half", "int8", "nms"]],
+        ["TensorFlow SavedModel", "saved_model", "_saved_model", True, True, ["batch", "data", "fraction", "int8", "keras", "nms"]],
         ["TensorFlow GraphDef", "pb", ".pb", True, True, ["batch"]],
         ["TensorFlow Lite", "tflite", ".tflite", True, False, ["batch", "data", "half", "int8", "nms", "fraction"]],
-        ["TensorFlow Edge TPU", "edgetpu", "_edgetpu.tflite", True, False, []],
-        ["TensorFlow.js", "tfjs", "_web_model", True, False, ["batch", "data", "half", "int8", "nms"]],
+        ["TensorFlow Edge TPU", "edgetpu", "_edgetpu.tflite", True, False, ["data", "fraction"]],
```
Comment (UltralyticsAssistant) [inline ultralytics/engine/exporter.py]: ⚠️ **HIGH**: `edgetpu` still won't pick up the documented default calibration dataset when `data` is omitted. The exporter only backfills `self.args.data` inside `if self.args.int8 and not self.args.data`, but `self.args.int8` is forced to `True` for Edge TPU later in the TensorFlow export block. That means `yolo export format=edgetpu` can reach `export_saved_model()` with `images=None`, so the new `data`/`fraction` support does not behave as documented unless the user passes `data` explicitly.

File: ultralytics/engine/exporter.py [hunk 2]
```diff
@@ -139,14 +139,14 @@ def export_formats():
             True,
             ["batch", "data", "dynamic", "half", "int8", "simplify", "nms", "fraction"],
         ],
-        ["CoreML", "coreml", ".mlpackage", True, False, ["batch", "data", "dynamic", "half", "int8", "nms"]],
-        ["TensorFlow SavedModel", "saved_model", "_saved_model", True, True, ["batch", "data", "int8", "keras", "nms"]],
+        ["CoreML", "coreml", ".mlpackage", True, False, ["batch", "dynamic", "half", "int8", "nms"]],
```
Comment (UltralyticsAssistant) [inline ultralytics/engine/exporter.py:142]: 💡 **MEDIUM**: Updating the `data` entries in `export_formats()` does not actually change argument enforcement. `validate_args()` only checks `half`, `int8`, `dynamic`, `keras`, `nms`, `batch`, and `fraction`, so `data=` is still silently accepted for formats like CoreML/MNN even after removing it here. If this PR is meant to align runtime behavior with the docs, `data` also needs to be part of the validated export args.

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Good cleanup overall, and the SavedModel/TF.js `fraction` additions line up with the existing TensorFlow export path. The two gaps above mean Edge TPU still won't honor the newly documented default calibration flow, and the `data` metadata changes are not enforced at runtime yet.

💬 Posted 2 inline comments

**Review by Laughing-q — APPROVED** [review]

**Comment by UltralyticsAssistant** [comment]
Merged! 🎉 Huge thanks to @lakshanthad for tightening up YOLO26 export support and documentation, and to @Laughing-q for the valuable contributions that helped bring this over the line.

“Accuracy is the twin brother of honesty; inaccuracy, of dishonesty.” — Nathaniel Hawthorne

That fits this PR perfectly: aligning exporter capabilities with the docs and tutorial tables makes INT8 calibration options clearer, reduces confusion, and gives users more confidence when exporting for TensorFlow-based, edge, and browser deployments. Really appreciate the thoughtful work to keep the code, docs, and user experience in sync.
